lib/ssl/src/ssl_handshake.erl

```diff
@@ -796,6 +838,14 @@ decode_handshake(Version, ?SERVER_HELLO, <<?BYTE(Major), ?BYTE(Minor), Random:32
        extensions = HelloExtensions};
 decode_handshake(_Version, ?CERTIFICATE, <<?UINT24(ACLen), ASN1Certs:ACLen/binary>>) ->
     #certificate{asn1_certificates = certs_to_list(ASN1Certs)};
+%% RFC 6066, servers return a certificate response along with their certificate
+%% by sending a "CertificateStatus" message immediately after the "Certificate"
+%% message and before any "ServerKeyExchange" or "CertificateRequest" messages.
+decode_handshake(_Version, ?CERTIFICATE_STATUS, <<?BYTE(StatusType),
```

Comment:
As `status_type` can only take one value, is it better to match on this value and let the last clause generate a `HANDSHAKE_FAILURE` if there is no match. This way we can remove the `get_ocsp_reponse/2` function.
```
enum { ocsp(1), (255) } CertificateStatusType;
```
```erlang
decode_handshake(_Version, ?CERTIFICATE_STATUS, <<?BYTE(?CERTIFICATE_STATUS_TYPE_OCSP), ...
```

Reply:
> As `status_type` can only take one value, is it better to match on this value and let the last clause generate a `HANDSHAKE_FAILURE` if there is no match. This way we can remove the `get_ocsp_reponse/2` function.
> 
> ```
> enum { ocsp(1), (255) } CertificateStatusType;
> ```
> 
> ```erlang
> decode_handshake(_Version, ?CERTIFICATE_STATUS, <<?BYTE(?CERTIFICATE_STATUS_TYPE_OCSP), ...
> ```

done